platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/internal/component/resolution/failure/ResolutionFailureHandlerIntegrationTest.groovy

```diff
@@ -371,24 +371,32 @@ class ResolutionFailureHandlerIntegrationTest extends AbstractIntegrationSpec {
         and: "Has error output"
         failure.assertHasDescription("Could not determine the dependencies of task ':forceResolution'.")
         failure.assertHasCause("Could not resolve all dependencies for configuration ':resolveMe'.")
-        assertFullMessageCorrect("""   > Found multiple transforms that can produce a variant of root project : with requested attributes:
+        assertFullMessageCorrect("""   > Found multiple transformation chains that produce a variant of 'root project :' with requested attributes:
        - color 'red'
        - shape 'round'
-     Found the following transforms:
-       - From 'configuration ':roundBlueLiquidElements'':
+     Found the following transformation chains:
+       - From configuration ':roundBlueLiquidElements':
            - With source attributes:
                - color 'blue'
                - shape 'round'
                - state 'liquid'
-           - Candidate transform(s):
-               - Transform 'BrokenTransform' producing attributes:
-                   - color 'red'
-                   - shape 'round'
-                   - state 'gas'
-               - Transform 'BrokenTransform' producing attributes:
-                   - color 'red'
-                   - shape 'round'
-                   - state 'solid'""")
+           - Candidate transformation chains:
+               - Transformation chain: 'BrokenTransform', producing attributes:
+                   - 'BrokenTransform':
+                       - Converts from attributes:
```

Comment:
Just an idea, since I can imagine these error messages getting quite long, is that we can combine the from and to attributes into a single line (if there is a matching from and to)

For example, instead of 
```
- Converts from attributes:
  - color 'blue'
  - state 'liquid'
  - attr `value`
- To attributes:
  - color 'red'
  - state 'gas'
```

Maybe something like
```
- Transforms color from 'blue' to 'red'
- Transforms state from 'liquid' to 'gas'
- Transforms from attr 'value'
```

Nothing we necessarily need to do in this PR, though I think it would help make this more concise.

Reply:
I thought about formatting corresponding attributes on the same line, but I think it isn't a good idea:
- Attribute values can be both very long and also of very different lengths, which could make sets of them formatted like this look uglier
- Seeing the just from attributes without to values is odd.  This format implies there should be a correspondence between from and to attributes, which is an idea I think we want to get away from
- I don't think being concise is the goal here.

If anything would help, I think adding an optional `description` when registering transforms would (#30785).  If that feature existed (and was used for a particular transform), we could just print it instead.

For instance:

```
registerTransform(Unzip) {
    from.attribute(type, 'zipped')
	from.attribute(color, 'blue')
    to.attribute(type, 'unzipped')
	description = "Unzip all blue files"
}
```

could skip the processing of attributes completely and just say something like:

```
- `Transform1`:
	- Converts from attributes:
	  - state 'liquid'
	- To attributes:
	  - state 'gas'
- `Unzip` (Unzip all blue files)
- `Transform2`:
	- Converts from attributes:
	  - attr 'val1'
	- To attributes:
	  - attr 'val2'
```